test/solvers/functional/functional_test.py

```diff
@@ -34,38 +34,65 @@
 PLACES = 8
 
 
-# TODO: maybe add tests for if translations etc. belongs to the wrong space
+# TODO: maybe add tests for if translations etc. belongs to the wrong space.
+# These tests doesn't work as intended now, since casting is possible between
+# spaces with the same number of discrete points.
 
+space_params = ['r10', 'uniform_discr']
+space_ids = [' space = {}'.format(p.ljust(10)) for p in space_params]
 
-# TODO: fixture for all functionals
-def test_derivative():
+
+@pytest.fixture(scope="module", ids=space_ids, params=space_params)
+def space(request, fn_impl):
+    name = request.param.strip()
+
+    if name == 'r10':
+        return odl.rn(10, impl=fn_impl)
+    elif name == 'uniform_discr':
+        # Discretization parameters
+        return odl.uniform_discr(0, 1, 7, impl=fn_impl)
+
+
+func_params = ['l1 ', 'l2', 'l2^2', 'constant']
+func_ids = [' f = {}'.format(p.ljust(10)) for p in func_params]
+
+
+@pytest.fixture(scope="module", ids=func_ids, params=func_params)
+def functional(request, space):
```

Comment:
I only added these because: 1) `ZeroFunctional` is a subclass of `ConstantFunctional`, 2) the indicator function has no derivative. But I can catch the `NotImplementedError` and take it as pass :)